spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -150,12 +150,7 @@ object Engine {
    * Runtime.getRuntime().availableProcessors() / 2
    */
   private def getCoreNumberFromProperty() = {
-    val coreNumber = System.getProperty("bigdl.coreNumber", getNumMachineCores.toString).toInt
-    if (coreNumber > getNumMachineCores) {
-      getNumMachineCores
-    } else {
-      coreNumber
-    }
+    System.getProperty("bigdl.coreNumber", getNumMachineCores.toString).toInt
```

Comment:
I think here should at least throw a warning...

I remove it because I think sometimes user want to use more thread to run the program.